includes/class-newspack-popups.php

```diff
@@ -376,6 +402,17 @@ public static function preview_post_permalink() {
 		return $recent_posts && count( $recent_posts ) > 0 ? get_the_permalink( $recent_posts[0] ) : '';
 	}
 
+	/**
+	 * Get preview archive permalink. Used to preview prompts in archive pages in the popup wizard.
+	 *
+	 * @return string
+	 */
+	public static function preview_archive_permalink() {
+		$categories = get_categories();
+
+		return count( $categories ) > 0 ? get_category_link( $categories[0] ) : '';
```

Comment:
On my local instance, `get_categories` returns an associative array, which results in a wrong offset error. After changing the `$categories` assignment to `array_values( get_categories() );` it's fine. Can you confirm that `get_categories` returns a regular array?

Reply:
Hum, I can confirm that `get_categories` returns a regular array
```
array (size=13)
  0 => 
    object(WP_Term)[3826]
      public 'term_id' => int 35
      public 'name' => string 'Arts' (length=4)
      public 'slug' => string '_newspack_arts' (length=14)
      public 'term_group' => int 0
      public 'term_taxonomy_id' => int 35
      public 'taxonomy' => string 'category' (length=8)
      public 'description' => string '' (length=0)
      public 'parent' => int 0
      public 'count' => int 3
      public 'filter' => string 'raw' (length=3)
      public 'cat_ID' => int 35
      public 'category_count' => int 3
      public 'category_description' => string '' (length=0)
      public 'cat_name' => string 'Arts' (length=4)
      public 'category_nicename' => string '_newspack_arts' (length=14)
      public 'category_parent' => int 0
  1 => 
    object(WP_Term)[3827]
      public 'term_id' => int 30
      public 'name' => string 'Entertainment' (length=13)
      public 'slug' => string '_newspack_entertainment' (length=23)
      public 'term_group' => int 0
      public 'term_taxonomy_id' => int 30
      public 'taxonomy' => string 'category' (length=8)
      public 'description' => string '' (length=0)
      public 'parent' => int 0
      public 'count' => int 4
      public 'filter' => string 'raw' (length=3)
      public 'cat_ID' => int 30
      public 'category_count' => int 4
      public 'category_description' => string '' (length=0)
      public 'cat_name' => string 'Entertainment' (length=13)
      public 'category_nicename' => string '_newspack_entertainment' (length=23)
      public 'category_parent' => int 0
...
```

Can you share a dump of `get_categories`, please? Adding a `array_values ` will fix it for your case and won't change mine, I just want to see how it looks.

Reply:
So wired, using `error_log` I can still see a regular array:
```
[
    {
        "term_id": 35,
        "name": "Arts",
        "slug": "_newspack_arts",
        "term_group": 0,
        "term_taxonomy_id": 35,
        "taxonomy": "category",
        "description": "",
        "parent": 0,
        "count": 3,
        "filter": "raw",
        "cat_ID": 35,
        "category_count": 3,
        "category_description": "",
        "cat_name": "Arts",
        "category_nicename": "_newspack_arts",
        "category_parent": 0
    },
    {
        "term_id": 30,
        "name": "Entertainment",
        "slug": "_newspack_entertainment",
        "term_group": 0,
        "term_taxonomy_id": 30,
        "taxonomy": "category",
        "description": "",
        "parent": 0,
        "count": 4,
        "filter": "raw",
        "cat_ID": 30,
        "category_count": 4,
        "category_description": "",
        "cat_name": "Entertainment",
        "category_nicename": "_newspack_entertainment",
        "category_parent": 0
    },
```
Anyway, I guess that an `array_value` won't hurt here!